backend/main.py

```
@@ -25,19 +26,29 @@
+
 async def discover_peers_task():
     while True:
-        discovery_message = '{"type": "DISCOVERY"}'
-        broadcast_address = '192.168.56.255'.encode('utf-8')
-        core_lib.send_udp_message(discovery_message.encode('utf-8'), broadcast_address, UDP_PORT)
+        discovery_payload = {
+            "type": "DISCOVERY",
+            "nickname": config.NICKNAME,
+            "version": config.PROTOCOL_VERSION,
+        }
+        discovery_message = json.dumps(discovery_payload)
```

Ruff (0.14.0)

Undefined name `json`

(F821)